continuous-integration/readme.md

```diff
@@ -1,3 +1,233 @@
+- [ ] **Pull request update(s) trigger staged releases**
+  - New commits related to a pull request should trigger a build / release into an integration environment. The production environment should be fully isolated from this process.
+- [ ] **Promote infrastructure changes across fixed environments**
+  - Infrastructure as code changes should be tested in a integration environment and promoted to all staging environment(s) then migrated to production with zero downtime for system users.
+- [ ] **Environment releases are triggered on merges into master**
+  - Code changes merged into the master branch should auto trigger a new release candidate for all cloud environments.
+  - One additional capability worth considering are automated rollbacks. Being able to revert commits into master and quickly go back to the last known good state(*previous feature release*) minimizes outages and downtime when bugs are introduced into environment(s).
+- [ ] **Testing in production**
+  - There are various [approaches](https://medium.com/@copyconstruct/testing-in-production-the-safe-way-18ca102d0ef1) with safely carrying out automated tests for production deployments. Some of these may include:
+    - Feature flagging
+    - A/B testing
+    - Traffic shifting
+
+## Developer Access to Latest Release Artifacts
+
+Our devops workflow should enable developers to get, install and run the latest system executable. Release executable(s) should be auto generated as part of our CI/CD pipeline(s).
+
+- [ ] **Developers can access latest executable**
+  - Latest system executable is available for all developers on the team. There should be a well-known place where developers can reference the release artifact.
+
+- [ ] **Release artifact is published for each git commit per open pull request or merges into master**
+
+## Integration Observability
+
+Applied state changes to the mainline build should be made available and communicated across the team. Centralizing logs and status(s) from build and release pipeline failures are essential for developers investigating broken builds.
+
+We recommend integrating Teams or Slack with CI/CD pipeline runs which helps keep the team continuously plugged into failures and build candidate status(s).
+
+- [ ] **Continuous integration top level dashboard**
+  - Modern CI providers have the capability to consolidate and report build status(s) within a given dashboard.
+  - Your CI dashboard should be able to correlate a build failure with a git commit.
+- [ ] **Build status badge in project readme**
+  - There should be a build status badge included in the root README of the project.
+- [ ] **Build notifications**
+  - Your CI process should be configured to send notifications to messaging platforms like Teams / Slack once the build completes. We recommend creating a separate channel to help consolidate and isolate these notifications.
+
+## Resources
```

Comment:
Please include a link to this repository as well: https://github.com/microsoft/terraform-provider-azuredevops